courses/models.py

```diff
@@ -234,6 +248,24 @@ def first_unexpired_run(self):
             lambda course_run: course_run.is_unexpired,
         )
 
+    @property
+    def unexpired_run_dates(self):
+        """
+        Gets all the start dates for unexpired CourseRuns associated with this Course
+
+        Returns:
+           [Datetime]: list of start_dates from all unexpired course runs
+        """
+        return list(
+            map(
+                lambda unexpired_course_run: unexpired_course_run.start_date,
+                filter(
+                    lambda course_run: course_run.is_unexpired,
+                    self.courseruns.all().order_by("start_date"),
+                ),
+            )
+        )
```

Comment:
please add unit test for this property.

Reply:
Currently WIP, tests and design need to be added yet.